src/include/config.h

```diff
@@ -0,0 +1,33 @@
+#ifndef IGRAPH_PRIVATE_CONFIG_H
+#define IGRAPH_PRIVATE_CONFIG_H
+
+#include "igraph_config.h"
+
+#define HAVE_STRCASECMP 1
+#define HAVE_STRNCASECMP 1
+/* #undef HAVE__STRICMP */
+/* #undef HAVE__STRNICMP */
+#define HAVE_STRDUP 1
+#define HAVE_STRNDUP 1
+// #define HAVE_USELOCALE 1
+/* #undef HAVE_XLOCALE */
+/* #undef HAVE__CONFIGTHREADLOCALE */
+
+#define HAVE_BUILTIN_OVERFLOW 1
+
+/* #undef HAVE__UMUL128 */
+/* #undef HAVE___UMULH */
+#define HAVE___UINT128_T 1
+
+#define HAVE_GLPK 1
+#define HAVE_LIBXML 1
+
+/* #undef INTERNAL_BLAS */
+/* #undef INTERNAL_LAPACK */
+/* #undef INTERNAL_ARPACK */
+/* #undef INTERNAL_GMP */
+
+#define IGRAPH_F77_SAVE static __thread
+#define IGRAPH_THREAD_LOCAL __thread
```

Comment:
This can be defined to be empty, and the `-DIGRAPH_THREAD_LOCAL=/**/` can be removed from the preprocessor flags (`CPPFLAGS`).

Reply:
I'm not sure what happened to `igraph_threading.h` here, it seems to be missing. It should contain

```
#define IGRAPH_THREAD_SAFE 0
```

This is also a generated file in C/igraph.

Reply:
> I'm not sure what happened to `igraph_threading.h` here, it seems to be missing. It should contain
> 
> ```
> #define IGRAPH_THREAD_SAFE 0
> ```
> 
> This is also a generated file in C/igraph.

https://github.com/igraph/rigraph/blob/igraph-0.10/src/include/igraph_threading.h this is also in include folder.